adserver/views.py

```diff
@@ -169,18 +175,23 @@ def get_context_data(self, **kwargs):
         report = AdvertiserReport(queryset)
         report.generate()
 
-        flights = (
-            Flight.objects.filter(campaign__advertiser=self.advertiser)
-            .filter(live=True)
-            .order_by("-live", "-end_date", "name")
-        )
+        flights = [
+            f
+            for f in (
+                Flight.objects.filter(campaign__advertiser=self.advertiser).order_by(
+                    "-live", "-end_date", "name"
+                )
+            )
+            if f.state in (FLIGHT_STATE_UPCOMING, FLIGHT_STATE_CURRENT)
```

Comment:
Hrm. Can we not do this with the queryset? `state__in= [FLIGHT_STATE_UPCOMING, FLIGHT_STATE_CURRENT]`?

Reply:
Ya, I guess that's probably better.

Reply:
Oh, no we can't. It's a calculated property.

Reply:
gotcha, this works then 👍